File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/recipes/domain/usecase/SaveRecipeUseCase.java
package com.tarbadev.witchcraft.recipes.domain.usecase;

import com.tarbadev.witchcraft.recipes.domain.entity.Recipe;
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository;
import org.springframework.stereotype.Component;

@Component
public class SaveRecipeUseCase {
  private final RecipeRepository recipeRepository;

  public SaveRecipeUseCase(RecipeRepository recipeRepository) {
    this.recipeRepository = recipeRepository;
  }

  public Recipe execute(Recipe recipe) {
    if (recipe.getId() != null && recipe.getId() > 0)
      recipe = recipeRepository.updateRecipe(recipe);
    else
      recipe = recipeRepository.saveRecipe(recipe);

    return recipe;
  }
}

File: WitchcraftApp/src/main/kotlin/com/tarbadev/witchcraft/recipes/domain/usecase/SaveRecipeUseCase.kt
package com.tarbadev.witchcraft.recipes.domain.usecase

import com.tarbadev.witchcraft.recipes.domain.entity.Recipe
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository
import org.springframework.stereotype.Component

@Component
class SaveRecipeUseCase(private val recipeRepository: RecipeRepository) {

    fun execute(recipe: Recipe): Recipe {
        return if (recipe.id > 0)
            recipeRepository.updateRecipe(recipe)
        else
            recipeRepository.saveRecipe(recipe)
    }
}
